### PR TITLE
MRG, ENH: Add scale bars to raw.plot

### PR DESCRIPTION
**Pull request by larsoner**
See below.

~~Eventually this can probably be extended to work for butterfly plots, but the overlay of grad+mag there might be a problem. (We could actually add one at the top and bottom in this case, but for now it's not implemented.)~~

~~Also at some point we might want to use the number formatting that #6210 fixed/used.~~

### CODE REVIEW

**Comment by larsoner** [comment]
Accidental opening of PR. But to give an idea, code:
```
$ mne browse_raw ~/mne_data/MNE-sample-data/MEG/sample/sample_audvis_filt-0-40_raw.fif
```
Shows (after scrolling down a bit):

Screenshot from 2019-06-04 16-40-45

And testing code:
```
import mne
import numpy as np

sfreq = 1000.
t = np.arange(10000) / sfreq
data = np.sin(2 * np.pi * 10. * t)
# +/- 1000 fT, 400 fT/cm, 20 uV
data = data * np.array([[1000e-15, 400e-13, 20e-6]]).T
info = mne.create_info(3, sfreq, ('mag', 'grad', 'eeg'))
raw = mne.io.RawArray(data, info)
fig = raw.plot()
```
Shows:

Screenshot from 2019-06-04 16-38-47

**Comment by agramfort** [comment]
a feature long overdue. Really nice !

>

**Comment by cbrnr** [comment]
Very nice! Here are some suggestions:

* Don't use black for the vertical bar and the label, because signal traces (at least for some types) and channel labels are by default also black. A dark gray color which is clearly different from black would look better.
* I wouldn't use a bold font for the scale bar label.
* Can you add a keyboard shortcut to enable/disable scale bars? Especially when I view many channels per page, scale bars can get in the way and it would be nice to have the option to hide them. The shortcut should also be mentioned in our Help window.
* Why is the scale bar not shown for STIM channels? Is this intentional?

**Comment by larsoner** [comment]
> Why is the scale bar not shown for STIM channels? Is this intentional?

Yes, because from the code it looked like the scale varies per stim channel and uses the max.

**Comment by jasmainak** [comment]
Super!

**Comment by larsoner** [comment]
@cbrnr keyboard option "s" added. I ended up going with a purple-ish color because it was easier to distinguish from the black.

File: mne/viz/raw.py [hunk 1]
```diff
@@ -998,24 +1005,36 @@ def _plot_raw_traces(params, color, bad_color, event_lines=None,
...
             # do NOT operate in-place lest this get screwed up
+
+            # apply user-supplied scale factor
             this_data = params['data'][inds[ch_ind]] * params['scale_factor']
-            this_color = bad_color if ch_name in info['bads'] else color
 
+            # clip to range (if relevant)
+            if params['clipping'] == 'transparent':
+                this_data[np.logical_or(this_data > 1,
+                                        this_data < -1)] = np.nan
```
Comment (cbrnr) [inline mne/viz/raw.py]: You could simplify with `np.abs(this_data) > 1`.

**Comment by cbrnr** [comment]
This looks great! Other than my minor suggestion +1 for merge!

**Comment by agramfort** [comment]
works like a charm except in the butterfly mode. Is this expected? if
I press "b" I don't see the scales anymore.

really nice !

**Comment by larsoner** [comment]
> works like a charm except in the butterfly mode. Is this expected?

Yes see top post:

> Eventually this can probably be extended to work for butterfly plots, but the overlay of grad+mag there might be a problem. (We could actually add one at the top and bottom in this case, but for now it's not implemented.)

**Comment by agramfort** [comment]
why is the overlay a problem?

>

**Comment by larsoner** [comment]
Ahh I thought that they mags and grads were actually shown together, on top of each other. But it looks like they are not. I'll see if it's an easy fix. If not it can be a follow-up PR

**Comment by larsoner** [comment]
Indeed it looks like removing the check for `not butterfly` was all that was needed for it to work. So we should be good there now

**Pull request merged**